File: examples/datasets/diff/new_code.js
// calculateSum adds two integers and returns their sum
function calculateSum(a, b) {
    const result = a + b;
    console.log(`Adding ${a} + ${b} = ${result}`);
    return result;
}

// calculateProduct multiplies two integers
function calculateProduct(a, b) {
    return a * b;
}

// Main execution
function main() {
    const sum = calculateSum(5, 3);
    const product = calculateProduct(5, 3);
    console.log(`Sum: ${sum}, Product: ${product}`);
}

// Run the main function
main();

File: examples/diff/new_code.go
package main

import "fmt"

// calculateSum adds two integers and returns their sum
func calculateSum(a, b int) int {
	result := a + b
	fmt.Printf("Adding %d + %d = %d\n", a, b, result)
	return result
}

// calculateProduct multiplies two integers
func calculateProduct(a, b int) int {
	return a * b
}

func main() {
	sum := calculateSum(5, 3)
	product := calculateProduct(5, 3)
	fmt.Printf("Sum: %d, Product: %d\n", sum, product)
}